programs/stake/src/stake_state.rs

```diff
@@ -362,43 +362,57 @@ impl Authorized {
     }
 }
 
+/// captures a rewards round as lamports to be awarded
+///  and the total points over which those lamports
+///  are to be distributed
+//  basically read as rewards/points, but in integers instead of as an f64
+pub struct PointValue {
+    pub rewards: u64, // lamports to split
+    pub points: u128, // over these points
+}
+
 impl Stake {
     pub fn stake(&self, epoch: Epoch, history: Option<&StakeHistory>) -> u64 {
         self.delegation.stake(epoch, history)
     }
 
     pub fn redeem_rewards(
         &mut self,
-        point_value: f64,
+        point_value: &PointValue,
         vote_state: &VoteState,
         stake_history: Option<&StakeHistory>,
     ) -> Option<(u64, u64)> {
         self.calculate_rewards(point_value, vote_state, stake_history)
-            .map(|(voters_reward, stakers_reward, credits_observed)| {
+            .map(|(stakers_reward, voters_reward, credits_observed)| {
                 self.credits_observed = credits_observed;
                 self.delegation.stake += stakers_reward;
-                (voters_reward, stakers_reward)
+                (stakers_reward, voters_reward)
             })
     }
 
-    /// for a given stake and vote_state, calculate what distributions and what updates should be made
-    /// returns a tuple in the case of a payout of:
-    ///   * voter_rewards to be distributed
-    ///   * staker_rewards to be distributed
-    ///   * new value for credits_observed in the stake
-    //  returns None if there's no payout or if any deserved payout is < 1 lamport
-    pub fn calculate_rewards(
+    pub fn calculate_points(
         &self,
-        point_value: f64,
         vote_state: &VoteState,
         stake_history: Option<&StakeHistory>,
-    ) -> Option<(u64, u64, u64)> {
+    ) -> u128 {
+        self.calculate_points_and_credits(vote_state, stake_history)
+            .0
+    }
+
+    /// for a given stake and vote_state, calculate how many
+    ///   points were earned (credits * stake) and new value
+    ///   for credits_observed were the points paid
+    pub fn calculate_points_and_credits(
+        &self,
+        vote_state: &VoteState,
+        stake_history: Option<&StakeHistory>,
+    ) -> (u128, u64) {
         if self.credits_observed >= vote_state.credits() {
-            return None;
+            return (0, 0);
```

Comment:
seems like I should have noticed `(..., 0)` can't be like this when reviewing and working on this.... ref: #13836

Reply:
as an execute, I remember I thought it's logically straight forward to translate `None` to `(0, 0)` on its own at the time. I should pay more attention for more surrounding context.